src/Content/Pager.php

```diff
@@ -0,0 +1,282 @@
+<?php
+
+namespace Friendica\Content;
+
+use Friendica\Core\L10n;
+
+/**
+ * The Pager has two very different output, Minimal and Full, see renderMinimal() and renderFull() for more details.
+ *
+ * @author Hypolite Petovan <mrpetovan@gmail.com>
+ */
+class Pager
+{
+	/**
+	 * @var integer
+	 */
+	private $page = 1;
+	/**
+	 * @var integer
+	 */
+	private $itemsPerPage = 50;
+
+	/**
+	 * @var string
+	 */
+	private $baseQueryString = '';
+
+	/**
+	 * Instantiates a new Pager with the base parameters.
+	 *
+	 * Guesses the page number from the GET parameter 'page'.
+	 *
+	 * @param string  $queryString  The query string of the current page
+	 * @param integer $itemsPerPage An optional number of items per page to override the default value
+	 */
+	public function __construct($queryString, $itemsPerPage = 50)
+	{
+		$this->setQueryString($queryString);
+		$this->setItemsPerPage($itemsPerPage);
+		$this->setPage(defaults($_GET, 'page', 1));
```

Comment:
just a suggestion (avoiding $_GET dependency inside class)

```php
function __construct($queryString, $page = 1, $itemsPerPage = 50)
{
....
$this->setPage($page);
}
```
And $page will be set during object initialization:
```php
$pager = new Pager($queryString, defaults($_GET, 'page'));
```

Reply:
I thought about it, and then I realized that it's actually self-contained since the Pager set the GET `page`  parameter in the query strings it produces. I actually like the current way better because the modules don't even have to know how the page parameter is called at all or how it is transmitted.